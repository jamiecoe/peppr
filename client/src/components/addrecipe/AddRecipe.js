import React, { Component } from 'react';
import { connect } from 'react-redux';

import UrlForm from './UrlForm';
import NewRecipeForm from './NewRecipeForm';
import Banner from '../MobileBanner';
import Navbar from '../Navbar';
import { showForm, resetRecipe } from '../../actions/recipes';
import { resetError } from '../../actions/auth';
import peppers from '../../assets/images/peppers.jpg';

export class AddRecipe extends Component {
  componentDidMount() {
    this.props.resetError();
    this.props.resetRecipe();
  }

  showRecipeForm = () => {
    this.props.showForm();
  }

  render() {
    return (
      <div>
        <Banner title="Add a recipe" />
        <img
          className="addrecipe__image"
          alt="red peppers on a dark background"
          src={peppers}
        />
        <div className="addrecipe__container">
          <h1 className="addrecipe__header addrecipe__innermargin">
            Add a Recipe
          </h1>
          <UrlForm />
          <button
            to="#"
            onClick={this.showRecipeForm}
            className="addrecipe__link addrecipe__innermargin"
          >
            Or enter your recipe details manually
          </button>
          {this.props.show ? <NewRecipeForm /> : <div />}
        </div>
        <Navbar page="addRecipe" />
      </div>
    );
  }
}

const mapStateToProps = state => ({ show: state.showForm });


export default connect(
  mapStateToProps,
  { showForm, resetError, resetRecipe },
)(AddRecipe);
