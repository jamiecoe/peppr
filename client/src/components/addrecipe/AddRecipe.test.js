import React from 'react';
import { shallow } from 'enzyme';
import { AddRecipe } from './AddRecipe';

describe('AddRecipes', () => {
  const mockShowForm = jest.fn();
  const mockResetError = jest.fn();
  const mockResetRecipe = jest.fn();

  const props = {
    showForm: mockShowForm,
    resetError: mockResetError,
    resetRecipe: mockResetRecipe,
    show: false,
  };

  let addRecipe = shallow(<AddRecipe {...props} />, {
    disableLifecycleMethods: false,
  });

  it('renders properly, and NOT show <NewRecipeForm>', () => {
    expect(addRecipe).toMatchSnapshot();
    expect(addRecipe.find('Connect(ReduxForm)').exists()).not.toBe(true);
  });

  it('should call `resetForm` and `resetRecipe` in componentDidMount()', () => {
    expect(mockResetError).toHaveBeenCalled();
    expect(mockResetRecipe).toHaveBeenCalled();
  });

  describe('when the user clicks on the `enter details manually` button', () => {
    beforeEach(() => {
      addRecipe
        .find('.addrecipe__link.addrecipe__innermargin')
        .simulate('click');
    });

    it('should call the `showForm` function supplied from props', () => {
      expect(mockShowForm).toHaveBeenCalled();
    });
  });

  describe('when `props.show` is true', () => {
    beforeEach(() => {
      props.show = true;
      addRecipe = shallow(<AddRecipe {...props} />);
    });

    it('should render properly, and show <NewRecipeForm>', () => {
      expect(addRecipe).toMatchSnapshot();
      expect(addRecipe.find('Connect(ReduxForm)').exists()).toBe(true);
    });
  });
});
