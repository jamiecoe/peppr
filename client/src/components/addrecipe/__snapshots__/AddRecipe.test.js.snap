// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddRecipes renders properly, and NOT show <NewRecipeForm> 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <AddRecipe
    resetError={[Function]}
    resetRecipe={[Function]}
    show={false}
    showForm={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Banner
          title="Add a recipe"
/>,
        <img
          alt="red peppers on a dark background"
          className="addrecipe__image"
          src="peppers.jpg"
/>,
        <div
          className="addrecipe__container"
>
          <h1
                    className="addrecipe__header addrecipe__innermargin"
          >
                    Add a Recipe
          </h1>
          <ReduxForm />
          <button
                    className="addrecipe__link addrecipe__innermargin"
                    onClick={[Function]}
                    to="#"
          >
                    Or enter your recipe details manually
          </button>
          <div />
</div>,
        <Connect(Navbar)
          page="addRecipe"
/>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "title": "Add a recipe",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alt": "red peppers on a dark background",
          "className": "addrecipe__image",
          "src": "peppers.jpg",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h1
              className="addrecipe__header addrecipe__innermargin"
>
              Add a Recipe
</h1>,
            <ReduxForm />,
            <button
              className="addrecipe__link addrecipe__innermargin"
              onClick={[Function]}
              to="#"
>
              Or enter your recipe details manually
</button>,
            <div />,
          ],
          "className": "addrecipe__container",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Add a Recipe",
              "className": "addrecipe__header addrecipe__innermargin",
            },
            "ref": null,
            "rendered": "Add a Recipe",
            "type": "h1",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Or enter your recipe details manually",
              "className": "addrecipe__link addrecipe__innermargin",
              "onClick": [Function],
              "to": "#",
            },
            "ref": null,
            "rendered": "Or enter your recipe details manually",
            "type": "button",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": "div",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "page": "addRecipe",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Banner
            title="Add a recipe"
/>,
          <img
            alt="red peppers on a dark background"
            className="addrecipe__image"
            src="peppers.jpg"
/>,
          <div
            className="addrecipe__container"
>
            <h1
                        className="addrecipe__header addrecipe__innermargin"
            >
                        Add a Recipe
            </h1>
            <ReduxForm />
            <button
                        className="addrecipe__link addrecipe__innermargin"
                        onClick={[Function]}
                        to="#"
            >
                        Or enter your recipe details manually
            </button>
            <div />
</div>,
          <Connect(Navbar)
            page="addRecipe"
/>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "title": "Add a recipe",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "red peppers on a dark background",
            "className": "addrecipe__image",
            "src": "peppers.jpg",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h1
                className="addrecipe__header addrecipe__innermargin"
>
                Add a Recipe
</h1>,
              <ReduxForm />,
              <button
                className="addrecipe__link addrecipe__innermargin"
                onClick={[Function]}
                to="#"
>
                Or enter your recipe details manually
</button>,
              <div />,
            ],
            "className": "addrecipe__container",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Add a Recipe",
                "className": "addrecipe__header addrecipe__innermargin",
              },
              "ref": null,
              "rendered": "Add a Recipe",
              "type": "h1",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Or enter your recipe details manually",
                "className": "addrecipe__link addrecipe__innermargin",
                "onClick": [Function],
                "to": "#",
              },
              "ref": null,
              "rendered": "Or enter your recipe details manually",
              "type": "button",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": "div",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "page": "addRecipe",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": false,
  },
}
`;

exports[`AddRecipes when \`props.show\` is true should render properly, and show <NewRecipeForm> 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <AddRecipe
    resetError={[Function]}
    resetRecipe={[Function]}
    show={true}
    showForm={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Banner
          title="Add a recipe"
/>,
        <img
          alt="red peppers on a dark background"
          className="addrecipe__image"
          src="peppers.jpg"
/>,
        <div
          className="addrecipe__container"
>
          <h1
                    className="addrecipe__header addrecipe__innermargin"
          >
                    Add a Recipe
          </h1>
          <ReduxForm />
          <button
                    className="addrecipe__link addrecipe__innermargin"
                    onClick={[Function]}
                    to="#"
          >
                    Or enter your recipe details manually
          </button>
          <Connect(ReduxForm) />
</div>,
        <Connect(Navbar)
          page="addRecipe"
/>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "title": "Add a recipe",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alt": "red peppers on a dark background",
          "className": "addrecipe__image",
          "src": "peppers.jpg",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h1
              className="addrecipe__header addrecipe__innermargin"
>
              Add a Recipe
</h1>,
            <ReduxForm />,
            <button
              className="addrecipe__link addrecipe__innermargin"
              onClick={[Function]}
              to="#"
>
              Or enter your recipe details manually
</button>,
            <Connect(ReduxForm) />,
          ],
          "className": "addrecipe__container",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Add a Recipe",
              "className": "addrecipe__header addrecipe__innermargin",
            },
            "ref": null,
            "rendered": "Add a Recipe",
            "type": "h1",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Or enter your recipe details manually",
              "className": "addrecipe__link addrecipe__innermargin",
              "onClick": [Function],
              "to": "#",
            },
            "ref": null,
            "rendered": "Or enter your recipe details manually",
            "type": "button",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "page": "addRecipe",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Banner
            title="Add a recipe"
/>,
          <img
            alt="red peppers on a dark background"
            className="addrecipe__image"
            src="peppers.jpg"
/>,
          <div
            className="addrecipe__container"
>
            <h1
                        className="addrecipe__header addrecipe__innermargin"
            >
                        Add a Recipe
            </h1>
            <ReduxForm />
            <button
                        className="addrecipe__link addrecipe__innermargin"
                        onClick={[Function]}
                        to="#"
            >
                        Or enter your recipe details manually
            </button>
            <Connect(ReduxForm) />
</div>,
          <Connect(Navbar)
            page="addRecipe"
/>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "title": "Add a recipe",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "red peppers on a dark background",
            "className": "addrecipe__image",
            "src": "peppers.jpg",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h1
                className="addrecipe__header addrecipe__innermargin"
>
                Add a Recipe
</h1>,
              <ReduxForm />,
              <button
                className="addrecipe__link addrecipe__innermargin"
                onClick={[Function]}
                to="#"
>
                Or enter your recipe details manually
</button>,
              <Connect(ReduxForm) />,
            ],
            "className": "addrecipe__container",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Add a Recipe",
                "className": "addrecipe__header addrecipe__innermargin",
              },
              "ref": null,
              "rendered": "Add a Recipe",
              "type": "h1",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Or enter your recipe details manually",
                "className": "addrecipe__link addrecipe__innermargin",
                "onClick": [Function],
                "to": "#",
              },
              "ref": null,
              "rendered": "Or enter your recipe details manually",
              "type": "button",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "page": "addRecipe",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
